src/app/calculator/calculator.component.ts

```diff
@@ -14,65 +15,160 @@ import { MOCK } from '../shared/mockdata';
 export class CalculatorComponent {
   constructor() {
     this.motocycles = MOCK.models;
-    this.canvasLeg = new p5(this.sketch, 'canvas-human');
+    this.canvasLeg = new p5(this.sketchLeg, 'canvas-human');
     this.canvasLeg.setup = () => {
-      this.canvasLeg.createCanvas(1000, 560);
+      this.canvasLeg.createCanvas(1000, 660).position(0, 0);
       this.canvasLeg.noLoop();
     };
+    this.canvasTors = new p5(this.sketchTorsAndArm, 'canvas-human');
+    this.canvasTors.setup = () => {
+      this.canvasTors.createCanvas(1000, 660).position(0, 0);
+      this.canvasTors.noLoop();
+    };
   }
 
+  canvasLeg;
+
+  canvasTors;
+
+  riderValues = {
+    height: null,
+    heightRiderPixel: null,
+    headPixel: null,
+    neckPixel: null,
+    torsPixel: null,
+    palmPixel: null,
+    armPixel: null,
+    legPixel: null,
+    waistToKneePixel: null,
+    kneeToFootPixel: null,
+    footPixel: null,
+    coordinateWaist: {
+      x: null,
+      y: null,
+    },
+    coordinateKnee: {
+      x: null,
+      y: null,
+    },
+    coordinateFootOnGround: {
+      x: null,
+      y: null,
+    },
+    coordinatePalmCenter: {
+      x: null,
+      y: null,
+    },
+    coordinateShoulder: {
+      x: null,
+      y: null,
+    },
+  };
+
+  motocycles: MotorcycleModel[];
+
   dataForm = new FormGroup({
-    motocycle: new FormControl('', Validators.required),
-    heightRider: new FormControl('', Validators.required),
-    legLength: new FormControl('', Validators.required),
+    motocycle: new FormControl('', this.validatorMotocycle),
+    heightRider: new FormControl('', this.validatorHeightRider),
   });
 
-  private canvasLeg;
+  validatorMotocycle(control: AbstractControl) {
+    if (!(control.value)) {
+      return {
+        isError: true,
+      };
+    }
+    return null;
+  }
 
-  motocycles: MotorcycleModel[];
+  validatorHeightRider(control: AbstractControl) {
+    if (!(control.value)) {
```

Comment:
and this part also you can change via Validators.require

Reply:
if I change to just Validators.require, then I still have to check the entered value so that it is not negative, not too big and not too small

Reply:
@LoshakovaAnna I talked about top part.